tests/zfs-tests/tests/functional/channel_program/lua_core/tst.stack_gsub.ksh

```diff
@@ -0,0 +1,33 @@
+#!/bin/ksh -p
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2020 by Delphix. All rights reserved.
+#
+
+. $STF_SUITE/tests/functional/channel_program/channel_common.kshlib
+
+#
+# DESCRIPTION:
+#       Overflowing the C stack using recursive gsub() should be handled
+#       gracefully.  gsub() uses more stack space than typical, so
+#	LUAI_MAXCCALLS must be set lower to ensure that we don't overflow
+#	the Linux kernel's stack.
+#
+
+verify_runnable "global"
+
+log_assert "recursive gsub() should be handled gracefully"
+
+log_mustnot_program $TESTPOOL $ZCP_ROOT/lua_core/tst.stack_gsub.zcp
+
```

Comment:
Given that the bug reported `zfs` command failing afterwards, does it make sense to run a simple `zfs` operation here that must succeed?

Reply:
That was because the "kernel panicked", meaning that the kernel killed the zfs module without rebooting.  In our system configuration the system would actually reboot if the kernel pancked.  In any case, `zfs program` would exit nonzero (as seen in the bug report) so this test would fail.